lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_playbook_method.rb

```diff
@@ -0,0 +1,112 @@
+module MiqAeEngine
+  class MiqAePlaybookMethod
+    PLAYBOOK_CLASS = ManageIQ::Providers::EmbeddedAutomationManager::ConfigurationScriptPayload
+
+    def initialize(aem, obj, inputs)
+      @workspace = obj.workspace
+      @inputs    = inputs
+      @aem       = aem
+    end
+
+    def run
+      if @workspace.persist_state_hash[method_key].present?
+        task_id = @workspace.persist_state_hash[method_key]
+        @aw = AutomateWorkspace.find_by(:guid => @workspace.persist_state_hash['automate_workspace_guid'])
+        check_task_status(task_id)
+      else
+        execute
+      end
+    end
+
+    private
+
+    def execute
+      @aw = AutomateWorkspace.create(:input  => serialize_workspace,
+                                     :user   => @workspace.ae_user,
+                                     :tenant => @workspace.ae_user.current_tenant)
+      playbook_options = build_options_hash
+      $miq_ae_logger.info("Playbook Method passing options to runner: #{playbook_options}")
+      begin
+        task_id = PLAYBOOK_CLASS.run(playbook_options)
+      rescue => err
+        $miq_ae_logger.error("Playbook Method Ended with error #{err.message}")
+        reset
+        raise MiqAeException::AbortInstantiation, err.message
+      end
+
+      @workspace.root['ae_state_started'].blank? ? wait_for_method(task_id) : check_task_status(task_id)
```

Comment:
Can you move `@workspace.root['ae_state_started'].blank?` into a method to make the intention is clear.

Something like
```ruby
def running_in_state_machine?
  @workspace.root['ae_state_started'].present?
end
```

Then this line is easier to read:
`running_in_state_machine? ? check_task_status(task_id) : wait_for_method(task_id)`